src/fields/nested-multi-select.ts

```diff
@@ -0,0 +1,83 @@
+import * as Yup from "yup";
+import { IFieldSchemaBase } from "../schema-generator";
+import { ERROR_MESSAGES } from "../shared";
+import { IFieldGenerator } from "./types";
+import { has } from "lodash";
+interface BaseOptionProps {
+	label: string;
+	key: string;
+}
+
+export type TNestedValues = IL1Value | IL2Value | IL3Value;
+export interface IL1Value {
+	[key: string]: IL2Value | string;
+}
+export interface IL2Value {
+	[key: string]: IL3Value | string;
+}
+export interface IL3Value {
+	[key: string]: string;
+}
+export interface TL1OptionProps extends BaseOptionProps {
+	value: string;
+	subItems?: TL2OptionProps[] | undefined;
+}
+export interface TL2OptionProps extends BaseOptionProps {
+	value: string;
+	subItems?: TL3OptionProps[] | undefined;
+}
+export interface TL3OptionProps extends BaseOptionProps {
+	value: string;
+	subItems?: undefined;
+}
+
+export interface INestedMultiSelectSchema<V = undefined> extends IFieldSchemaBase<"nested-multi-select", V> {
+	options: TL1OptionProps[];
+}
```

Comment:
- there are many typings declared, can move them to a `types.ts` file for better readability
- interfaces and types should be prepended by `I` and `T` respectively
- for L1 and L2 option props, they can come with either a `value` or `subItems` but never both, current typing allows both, you can prevent this via the union type, e.g.

```
interface IBaseOptionPropsWithValues extends BaseOptionProps {
	value: string;
	subItems?: never;
}

interface IBaseOptionPropsWithSubItems<T> extends BaseOptionProps {
	value?: never;
	subItems: T | undefined;
}

export type TL1OptionProps = IBaseOptionPropsWithValues | IBaseOptionPropsWithSubItems<TL2OptionProps[]>;
```

Reply:
thanks for the comments, updated the pr based on your suggestions. 
the typings were probably remnants of my previous implementation, sorry bout that 😓